content-src/components/DiscoveryStreamComponents/DSLinkMenu/DSLinkMenu.jsx

```diff
@@ -0,0 +1,84 @@
+import {FormattedMessage, injectIntl} from "react-intl";
+import {LinkMenu} from "content-src/components/LinkMenu/LinkMenu";
+import React from "react";
+
+export class _DSLinkMenu extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeCard: null,
+      showContextMenu: false,
+    };
+    this.onMenuButtonClick = this.onMenuButtonClick.bind(this);
+    this.onMenuUpdate = this.onMenuUpdate.bind(this);
+    this.onMenuShow = this.onMenuShow.bind(this);
+    this.contextMenuButtonRef = React.createRef();
+  }
+
+  onMenuButtonClick(event) {
+    event.preventDefault();
+    this.setState({
+      activeCard: this.props.index,
+      showContextMenu: true,
+    });
+  }
+
+  onMenuUpdate(showContextMenu) {
+    this.setState({showContextMenu});
+  }
+
+  onMenuShow() {
+    if (window.scrollMaxX > 0) {
+      this.contextMenuButtonRef.current.parentElement.classList.add("last-item");
```

Comment:
@Mardak Updated logic to add last-item className by checking if rendering of context menu is causing any horizontal scrolls, if yes we should open context menu to left.

Reply:
Can you help me understand why this is causing horizontal scroll in parts of this patch?

Reply:
By default context-menu open to the right when clicked. For last item in a row, this default open behavior causes horizontal scroll and need to be handled by fliping context menu direction to the left.

Reply:
Ah, interesting. For what it's worth this is an issue in topsites too.

Reply:
If I open a menu in topsites with discovery turned off, it somehow also manages to flip it to the left.

Reply:
I would be curious to see how this gets solved in the old activity stream, and why it's no longer working in discovery stream, including topsites.

Reply:
https://searchfox.org/mozilla-central/search?q=context-menu-open-left&path=

we are handling explicit breakpoints in css. See context-menu-open-left in _sections.scss for recommended top stories and _topsites.scss for top sites.

Considering the dynamic layout of discovery stream experience, IMO its hard to handle different edge cases in css

Reply:
I'm seeing last-item applied to the middle or second element in 3 or 4 card grids.

Reply:
Oh, I see, it moves it if there is a horizontal scroll bar, doesn't seem to matter if it caused it or not.

Reply:
I can make the menu appear partially off screen on the left side. (getting a screen shot)

Reply:
may be rename last-item to last-screen-item or something better to explain it's last item closer to edge on the screen, so if a layout has middle or second element on the edge it will have last-screen-item applied

Reply:
> I can make the menu appear partially off screen on the left side. (getting a screen shot)

With responsive support, horizontal scroll appearing due to window resize should be resolved